server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -178,31 +184,55 @@ public void onNewInfo(ClusterInfo info) {
             final boolean isWarmNode = REMOTE_CAPABLE.equals(getNodePool(routingNode));
             nodeDiskEvaluator.setNodeType(isWarmNode);
 
-            if (nodeDiskEvaluator.isNodeExceedingFloodStageWatermark(usage)) {
+            if (isWarmNode) {
+                AggregateFileCacheStats aggregateFileCacheStats = info.getNodeFileCacheStats().getOrDefault(usage.getNodeId(), null);
```

Comment:
Can we have some setting to turn on/off this feature if something not working as expected.